application/Common/Shared/Utility/NativeMethods.cs

```diff
@@ -11,6 +11,16 @@ namespace MORR.Shared.Utility
     /// </summary>
     public static class NativeMethods
     {
+        #region Properties
+
+        /// <summary>
+        /// Defines whether a current message loop should be cancelled.
+        /// We use a bool type for this as it is atomic and should not be problematic concerning threading.
```

Comment:
I may be missing something, but i don't see why this needs to be atomatic. The only situation where this comes into place is when different threads try to start and stop the loop at roughly the same time, and if this happens, the result is unpredictable either way.

Just a question out of curiosity, doesn't need implementation change.